### ISSUE TITLE
release 4.0.5 to maven central (or another publicly accessible repo)

### ISSUE THREAD

**Issue by afeinberg** [paultuckey/urlrewritefilter#191]
4.0.5 contains a fix for qsappend (issue 116), but it isn't available anywhere. can 4.0.5 be pushed to a public repo?

**Comment by wbazant** [comment]
@paultuckey are you the only maintainer of this repository? Is there anyone else allowed to do releases of new versions?

**Comment by paultuckey** [comment]
sure happy for someone to do it.  the only reason I haven't is that I haven't had the time to setup many app servers and run the full test suite over them all.

**Comment by tubbynl** [comment]
@paultuckey i do think it's sensible to setup the release proces anyway

for running the full testsuite we could start with a single application server and work our way through. i do think that in a minimal setting being able to release is valuable on it's own.

either way extra maintainers ( @wbazant ?) will be required to setup some sort of CI if @paultuckey doesn't have the time.

**Comment by chadlwilson** [comment]
Recently 5.x versions have been released with include migration to the `jakarta.x` namespaces - thanks for the work on this @paultuckey !

However there seem to be a number of changes/fixes that were never released for the `4.x` stream over many years, including #116 mentioned here, but also #182. 

If a build/release process is now set up, would it be possible to get a `4.x` release with these fixes, given the `5.x` change is a big breaking change that typically requires synchronised migration of all sorts of frameworks/libraries etc (web server/servlet container, web framework and more)?

Due to the lack of git tags, it's a bit difficult to find appropriate start and end points but the changes up until when the POM was changed to `5.0.0-SNAPSOT` seem to be https://github.com/paultuckey/urlrewritefilter/compare/0dc4fd1de725f93598b7015fd0217947fb56542d...4d9d29d 

There were a number of changes/fixes subsequent to the POM major version change in 2019 that seem to be relevant to 4.x as well, and perhaps could be cherry-picked.

**Comment by paultuckey** [comment]
Yep, I've now got a release process working ok.  If anyone wants to submit a PR backporting changes to 4.x I'm happy to accept it.  I can just make a long lived branch for 4.x.

**Comment by chadlwilson** [comment]
That would be great (to pick an appropriate `4.x` branch starting point from your perspective)! I imagine at a minimum some of the more Maven-oriented build/release and POM changes will definitely be needed to be cherrypicked to allow easy release.

I haven't really dug through to understand which of the changes after https://github.com/paultuckey/urlrewritefilter/commit/ffd9159f3a8c229594e75b1c26d809cddcc7cc52 were intentionally "breaking" as a v5 vs not-necessarily-breaking, as when the POM was initially bumped it didn't seem at the time motivated by a specific breaking change - more forward looking.

Of the PRs merged, if it's literally only the `jakarta.servlet` plus servlet-api 5.x changes that are truly **breaking** in `5.x` perhaps that's easier because one can probably cherry pick basically everything except that change.

**Comment by paultuckey** [comment]
Yes that is a good commit to use.  I think the changes related to releasing will be easy to copy in.  The changes around container testing will not, so I'd probably skip those.

https://github.com/paultuckey/urlrewritefilter/tree/4.x
